administrator/components/com_admin/script.php

```diff
@@ -310,6 +312,49 @@ private function migrateCompatPlugin($rowOld)
         )->execute();
     }
 
+    /**
+     * This method is for migration for old logrotation system plugin migration to task.
+     *
+     * @param   \stdClass  $data  Object with the extension's record in the `#__extensions` table
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    private function migrateLogRotationPlugin($data)
+    {
+        /** @var SchedulerComponent $component */
+        $component = Factory::getApplication()->bootComponent('com_scheduler');
+
+        /** @var TaskModel $model */
+        $model = $component->getMVCFactory()->createModel('Task', 'Administrator', ['ignore_request' => true]);
+
+        if (!$data->enabled) {
+            return;
+        }
+
+        // Get the timeout, as configured in plg_system_logrotation
+        $params       = new Registry($data->params);
+        $cachetimeout = (int) $params->get('cachetimeout', 6);
+        $lastrun      = (int) $params->get('lastrun', 0);
+
+        $task = [
+            'title'           => 'RotateLogs',
+            'type'            => 'rotation.logs',
+            'execution_rules' => [
+                'rule-type'      => 'interval-hours',
```

Comment:
```suggestion
                'rule-type'      => 'interval-days',
```

System plugin saves the information in days not in hours

Reply:
Would also need to be changed in extensions.sql when inserting the new task for new installations, e.g. here for MySQL:

https://github.com/joomla/joomla-cms/pull/40519/files#diff-fa733b616bbcc3e559d5003d630a5ba449c1cec34c00d1f44aa6d6c7d1dc75d0R929

Reply:
in mysql it's already days

Reply:
ah yes i just see.